src/features/bulk/store/slice.ts

```diff
@@ -0,0 +1,28 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { IBulkMetadataState, IBulkMetadataFile } from './model';
+
+export const initialState: IBulkMetadataState = {
+	currentStep: 'upload',
+	files: [],
+	fields: [],
+};
+
+export const slice = createSlice({
+	name: 'dmsbulkupload',
+	initialState,
+	reducers: {
+		setFile: (state: IBulkMetadataState, action: PayloadAction<IBulkMetadataFile>) => {
+			const newFile: IBulkMetadataFile = {
+				id: action.payload.id,
+				url: action.payload.url,
+				uploadedFile: action.payload.uploadedFile,
+				metadata: action.payload.metadata
```

Comment:
I saw that in the interface "metadata" is not mandatory. Shouldn't it be checked if the property exists, or else use a default value?

Reply:
Good find!